src/Nethermind/Nethermind.JsonRpc/Modules/DebugModule/DebugRpcModule.cs

```diff
@@ -157,6 +215,13 @@ public ResultWrapper<GethLikeTxTrace> debug_traceTransactionInBlockByHash(byte[]
 
     public ResultWrapper<GethLikeTxTrace> debug_traceTransactionInBlockByIndex(byte[] blockRlp, int txIndex, GethTraceOptions options = null)
     {
+        Block block = _blockDecoder.Decode(blockRlp);
```

Comment:
If we decode RLP here, would be good to pass it to debugbridge rather than decoding it there 2nd time

Reply:
Yes, that's also what I was referring to as some operations that are being done twice now. Though in this case we'll require a change to the `GethStyleTracer` API though, as we can't just pass a `Block` to it - I assume that's okay